### PR TITLE
ISSUE-47: refactor to ZipArchive and bump dependencies

### PR DESCRIPTION
**Pull request by DiegoPino**
# Overview

See #47 

This pull removes external library dependencies for ZIP management. Since most of the operations done are quite basic using PHP `ZipArchive` seems like a simpler future proof solution.

Other changes:
- Presence of `zipArchive` class is done via a static method on the `Package` Class. Its a bit strict since some methods could work without ZIP access but it hard to assume people could make real use of the library without it.
- PHP 7.1 as base requirement. Keeping up with 5.6 seems like a complex idea and also `frictionlessdata/tableschema` goes in that direction
- There was a tiny bug in `Utils::removeDir` that was assuming all paths would be a `dir`
- Test were updated
- General (first pass) Coding Standards and some `PHPDOCs` updates. This will require a few more passes to be completely compliant (I need to learn more about the package to make sure type hinting of arguments is consistent with the callers and the docs)
- Some type hints I could check were actually correct in terms of the method callers/implementations.

**Note:** All tests are passing in PHP 7.1, 7.2 and 7.3 but manual testing is still needed.

Since the changes here may feel like larger, I would propose this could be a new 0.2 version.

Thanks a lot, please let me know if something makes no sense and needs refactor and thanks again for this wonderful library

---

Please preserve this line to notify @OriHoch (lead of this repository)

### CODE REVIEW

**Comment by DiegoPino** [comment]
@OriHoch CI is failing for obvious reasons. Wonder if you would be OK if i move from `php-cs-fixer` to `phpcs` to also extend the testing to PHP 7.4. Issue is that `php-cs-fixer` is only compatible until PHP 7.1 and 7.1 is already EOL. 

If yes. Would be going `--standard=psr2` in conformance with this project's coding standards?
(checked and its almost there, just a few things that can be fixed automagically via `phpcbf`)
I can update Travis, etc if that is the case. Please let me know, I have a few hours tomorrow and would like to give this a full refresh and bump also PHP combat up.

Pushed some CI ideas

Thanks

File: src/Factory.php [hunk 1]
```diff
@@ -338,29 +343,37 @@ protected static function isFileZipSource($source)
 
     protected static function loadHttpZipSource($source)
     {
+        print_r('calling loadHttpZipSource');
```
Comment (OriHoch) [inline src/Factory.php]: Please remove this print

File: src/Package.php [hunk 2]
```diff
@@ -24,4 +53,15 @@ public static function create($descriptor = null, $basePath = null)
 
         return new $packageClass($descriptor, $basePath, true);
     }
+
+  /**
+   * @throws \Exception
+   */
+    private static function isZipPresent()
```
Comment (OriHoch) [inline src/Package.php]: Could we throw this exception only if zip support is actually needed?

**Review by OriHoch — COMMENTED** [review]
Looks good 👍 thanks
See comments

File: src/Factory.php (hunk 1 above)
Comment (DiegoPino) [reply src/Factory.php]: Oh, so sorry. Gosh. Addressed and committed

File: src/Package.php (hunk 2 above)
Comment (DiegoPino) [reply src/Package.php]: @OriHoch thanks. Yes of course. Do you prefer to have it at a higher level or directly on methods that use the ZIP? Basically during source detection or in the actual Package/ZIP methods?

Comment (OriHoch) [reply src/Package.php]: I don't have a preference, whatever makes sense for you

**Comment by DiegoPino** [comment]
@OriHoch sorry for the delay here. Got distracted with our own many GitHub issues and upcoming releases. 
Now `\frictionlessdata\datapackage\Package::isZipPresent`` is called on in methods that actually need it, sometimes a bit earlier to avoid generate Temporaries. Tests passed and Coding Standards (PSR-2) too. I did a lot of PHPDocs indenting so that makes a bulk of this

In any case the composer.json is already stating that ext-zip (same as the old `chumper/zipper` did) so its quite impossible to deploy with out zip support, but you never know (like partial zip support, where people forget to copy the .so in place).

Hope I got it right this time! Thanks a lot for your time and patience

**Comment by OriHoch** [comment]
looks great, thanks!

**Pull request merged**